pkg/operator/ceph/cluster/osd/osd.go

```diff
@@ -1002,44 +1015,79 @@ func (c *Cluster) updateCephStorageStatus() error {
 		cephClusterStorage.OSD.MigrationStatus.Pending = len(migrationConfig.osds)
 	}
 
-	err = c.context.Client.Get(c.clusterInfo.Context, c.clusterInfo.NamespacedName(), &cephCluster)
-	if err != nil {
-		if kerrors.IsNotFound(err) {
-			logger.Debug("CephCluster resource not found. Ignoring since object must be deleted.")
-			return nil
+	err = retry.RetryOnConflict(retry.DefaultRetry, func() error {
+		cephCluster := cephv1.CephCluster{}
+		err := c.context.Client.Get(c.clusterInfo.Context, c.clusterInfo.NamespacedName(), &cephCluster)
+		if err != nil {
+			if kerrors.IsNotFound(err) {
+				logger.Debug("CephCluster resource not found. Ignoring since object must be deleted.")
+				return nil
+			}
+			return errors.Wrapf(err, "failed to retrieve ceph cluster %q to update ceph Storage", c.clusterInfo.NamespacedName().Name)
 		}
-		return errors.Wrapf(err, "failed to retrieve ceph cluster %q to update ceph Storage", c.clusterInfo.NamespacedName().Name)
-	}
-	if !reflect.DeepEqual(cephCluster.Status.CephStorage, cephClusterStorage) {
```

Comment:
@sp98 this check was removed to help simplify the code here. I don't think the check is totally necessary, but please double check my logic here